### PR TITLE
fix: fix OAuth error response

### PR DESCRIPTION
**Pull request by aecra**

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Steve0x2a @seriouszyx plz review

File: object/token.go [hunk 1]
```diff
@@ -311,59 +321,40 @@ func GetOAuthCode(userId string, clientId string, responseType string, redirectU
...
 	if !IsGrantTypeValid(grantType, application.GrantTypes) && tag == "" {
-		errString = fmt.Sprintf("error: grant_type: %s is not supported in this application", grantType)
-		return &TokenWrapper{
-			AccessToken: errString,
-			TokenType:   "",
-			ExpiresIn:   0,
-			Scope:       "",
-			Error:       errString,
+		return &TokenError{
+			Error: UNSUPPORTED_GRANT_TYPE,
```
Comment (seriouszyx) [inline object/token.go:336]: add `ErrorDescription`?

File: object/token.go [hunk 2]
```diff
@@ -380,81 +371,56 @@ func GetOAuthToken(grantType string, clientId string, clientSecret string, code
...
 	if application == nil {
-		errString = "error: invalid client_id"
-		return &TokenWrapper{
-			AccessToken: errString,
-			TokenType:   "",
-			ExpiresIn:   0,
-			Scope:       "",
-			Error:       errString,
+		return &TokenError{
+			Error: INVALID_CLIENT,
```
Comment (seriouszyx) [inline object/token.go:385]: add `ErrorDescription` to provide more detailed error messages anywhere

**Comment by hsluoyz** [comment]
@seriouszyx

**Review by seriouszyx — APPROVED** [review]

**Pull request merged**